cfme/services/catalogs/catalog.py

```diff
@@ -86,24 +81,100 @@ def create(self):
         flash.assert_success_message(flash_str.format(self.name))
 
     def update(self, updates):
-        sel.force_navigate('catalog_edit', context={'catalog': self})
+        navigate_to(self, 'Edit')
         web_ui.fill(form, {'name_text': updates.get('name', None),
                            'description_text': updates.get('description', None),
                            'button_multiselect': updates.get('items', None)},
                     action=form.save_button)
         flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
 
     def delete(self):
-        sel.force_navigate('catalog', context={'catalog': self})
-        cfg_btn("Remove Item from the VMDB", invokes_alert=True)
+        navigate_to(self, 'Details')
+        cfg_btn(version.pick(cfg_details_remove), invokes_alert=True)
         sel.handle_alert()
         flash.assert_success_message(
             'Catalog "{}": Delete successful'.format(self.description or self.name))
 
     @property
     def exists(self):
         try:
-            sel.force_navigate('catalog', context={'catalog': self})
+            navigate_to(self, 'Details')
             return True
-        except CandidateNotFound:
+        # web_ui.Table.click_row_by_cells throws a NameError exception on no match
+        except NameError:
             return False
+
+
+@navigator.register(Catalog, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Services', 'Catalogs')(None)
+        tree = accordion.tree('Catalogs')
+        tree.click_path('All Catalogs')
+
+    def resetter(self):
+        # Default list view
+        tb.select('List View')
+        if paginator.page_controls_exist():
+            # Make sure nothing is selected
+            sel.check(paginator.check_all())
+            sel.uncheck(paginator.check_all())
+
+    def am_i_here(self):
+        active = accordion.is_active('All Catalogs')
+        header = sel.is_displayed('//h1//span[@id="explorer_title_text" and text()="All Catalogs"]')
+        return active and header
+
+
+@navigator.register(Catalog, 'Add')
+class Add(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Add a New Catalog')
+
+    def am_i_here(self):
```

Comment:
I don't know for this certain form but I woudn't recommend to use _am_i_here for our forms. 
1. I think this method will be used really rarely;
2. There are dependencies between controls states, values, etc. next test would fail if it gets a dialog with state from previous test.